### PR TITLE
Fix error on Treasure Table "update" and "add" + visual fixes

### PR DESCRIPTION
**Pull request by freohr**
Quick fix for the "Treasure Table" problems encountered in #80

### CODE REVIEW

**Comment by anthonyronda** [comment]
This doesn't fix it for me, I'm afraid. I'm experiencing the same behavior in this pull request branch as on main branch, the error described in issue 80.

In particular, I can confirm that the validation function that's causing the error is blocking the table entry from writing to the table *before* any of the code committed here is run. If you're experiencing otherwise, there may be a race condition.

**Review by anthonyronda — CHANGES_REQUESTED** [review]
Does not work on my machine, error #80 persists

**Comment by anthonyronda** [comment]
As a follow up to this, we discovered some problems with a test build based on main branch that had not surfaced on my dev builds. I'll revisit this PR after we've gotten that resolved

File: src/module/treasure.js [hunk 1]
```diff
@@ -1,32 +1,41 @@
 export const augmentTable = (table, html, data) => {
   // Treasure Toggle
+  const isTreasureTable = table.object.getFlag("ose", "treasure");
```
Comment (anthonyronda) [inline src/module/treasure.js:3]: getFlag() returns undefined for a table where the flag has not yet been created by the click event

File: src/module/treasure.js [hunk 2]
```diff
@@ -1,32 +1,41 @@
...
+
+  let treasureTableToggle = $("<div class='toggle-treasure' title='Toggle Treasure Table'></div>");
+  treasureTableToggle.toggleClass("active", isTreasureTable);
```
Comment (anthonyronda) [inline src/module/treasure.js:6]: [.toggleClass( className, state )](https://api.jquery.com/toggleClass/#toggleClass-className-state)

    className
    Type: [String](http://api.jquery.com/Types/#String)
    One or more classes (separated by spaces) to be toggled for each element in the matched set.
    state
    Type: [Boolean](http://api.jquery.com/Types/#Boolean)
    A boolean (not just truthy/falsy) value to determine whether the class should be added or removed.

While we're giving undefined for the state, it appears as though toggleClass treats it as true with no error returned. This results in unexpected behavior: it appears to default to toggled as active even though there's no flag by default.

**Review by anthonyronda — CHANGES_REQUESTED** [review]
Sorry I found another nitpick!

The toggle button is defaulting to style class `active` on tables without a `treasure` flag. I think the expected behavior is to not show as `active` by default, while still not adding the `treasure` flag unless the treasure button is clicked.

I left comments to show why this is happening.

File: src/module/treasure.js (hunk 1 above)
Comment (freohr) [reply src/module/treasure.js:3]: Good point, I didn't think of that either (still need to get used to the joy of `undefined` coming from C++). Would `getFlag() || false` enforce a boolean value?
This should be the only place where we change this flag for this Table object, but I don't know what would happen if something else put a flag that wasn't a boolean value on this object with this name.

Comment (freohr) [reply src/module/treasure.js:3]: I just thought of it, but is it an accepted practice if I use `!!getFlag()` to force a boolean value?

Comment (anthonyronda) [reply src/module/treasure.js:3]: I think you can Boolean(getFlag()) too. I don't mind either way

File: src/module/treasure.js (hunk 2 above)
Comment (freohr) [reply src/module/treasure.js:6]: I understand, I intended to be a bit smarter than a ternary with 2 raw strings, and take advantage of JQuery for clarity. I put my questions regarding the handling of the `active` var above.

**Review by anthonyronda — APPROVED** [review]
Works for me now

Lets Get This Merged!

**Pull request merged**